docs/enhancements/infrastructure-monitoring-service-api/alarms.md

```diff
@@ -0,0 +1,668 @@
+```yaml
+title: lifecycle-of-infrastructure-monitoring-alarms-api
+authors:
+  - @browsell
+  - @Jennifer-chen-rh
+  - @pixelsoccupied
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - TBD
+approvers: # A single approver is preferred, the role of the approver is to raise important questions, help ensure the enhancement receives reviews from all applicable areas/SMEs, and determine when consensus is achieved such that the EP can move forward to implementation.  Having multiple approvers makes it difficult to determine who is responsible for the actual approval.
+  - @browsell
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - TBD
+creation-date: 2024-08-26
+last-updated: yyyy-mm-dd
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - TBD
+see-also:
+  - "None"
+replaces:
+  - "None"
+superseded-by:
+  - "None"
+```
+
+# Lifecycle of infrastructure monitoring alarms api
+
+# Table of Contents
+- [Summary](#Summary)
+- [Goals](#Goals)
+- [Key o-ran data structures](#key-o-ran-data-structures)
+- [InfrastructureMonitoring Service API](#Infrastructure-Monitoring-Service-Alarms-API)
+- [Database schema](#schema)
+- [Init behaviour](#init)
+- [Ready behaviour](#ready)
+  - [Find AlarmDefinitionID and ProbableCauseID from current Alerts](#for-a-given-resourcetypeid-and-alarmname-coming-from-am-alert-find-the-alarmdefinitionid-and-probablecauseid)
+  - [Notification tracking](#notification-tracking)
+  - [Cleaning historical data](#daily-archive-cleanup)
+- [Kubernetes](#k8s-resources)
+- [Tooling](#tooling-and-general-dev-guidelines)
+- [Future Updates](#future-updates)
+
+## Summary
+
+`o-ran` requires `InfrastructureMonitoring Service Alarms API` which is a collection of APIs that can be queried by client to 
+monitor the health of the `o-cloud`. This enhancement describes initialization steps and ready steps for `InfrastructureMonitoring Service Alarms API` 
+as well everything that's needed to fully develop the app.
+
+At a high level, this service can be viewed as a thin wrapper of ACM observability stack which translates 
+OCP cluster resources to data structures understood and defined by `o-ran` spec. Among other things
+the service exposes APIs, configures Alertmanager deployment, read PrometheusRules from managedclusters and finally 
+store data in a persistent storage. 
+
+### Goals
```

Comment:
Another goal is to allow future integration of alarms from additional sources (i.e. H/W)

Reply:
Updated